opencast-backend/annotation-impl/src/main/java/org/opencast/annotation/impl/persistence/ExtendedAnnotationServiceJpaImpl.java

```diff
@@ -83,28 +82,47 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import javax.persistence.EntityManager;
 import javax.persistence.EntityManagerFactory;
 import javax.persistence.NoResultException;
 import javax.persistence.NonUniqueResultException;
-import javax.persistence.Query;
 import javax.persistence.RollbackException;
+import javax.persistence.TypedQuery;
 
 /**
  * JPA-based implementation of the {@link ExtendedAnnotationService}.
  */
 @Component
 public final class ExtendedAnnotationServiceJpaImpl implements ExtendedAnnotationService {
 
-  private PersistenceEnv persistenceEnv;
+  private EntityManagerFactory entityManagerFactory;
+  private DBSessionFactory dbSessionFactory;
+  private DBSession db;
   private SecurityService securityService;
   private AuthorizationService authorizationService;
   private SearchService searchService;
 
+  @Activate
+  public void activate() {
+    db = dbSessionFactory.createSession(entityManagerFactory);
+  }
+
+  @Deactivate
+  public synchronized void deactivate() {
+    db.close();
+  }
+
   @Reference(target = "(osgi.unit.name=org.opencast.annotation.impl.persistence)")
   public void setEntityManagerFactory(EntityManagerFactory entityManagerFactory) {
-    this.persistenceEnv = PersistenceEnvs.persistenceEnvironment(entityManagerFactory);
+    this.entityManagerFactory = entityManagerFactory;
+  }
+
+  @Reference
+  public void setDBSessionFactory(DBSessionFactory dbSessionFactory) {
+    this.dbSessionFactory = dbSessionFactory;
```

Comment:
Apparently not, can't explain why though.

Reply:
OK? How do you know, though? 😅

Reply:
Because that's the way it is in Opencast and that hasn't burned up in flames yet.

Reply:
Doing something like

```suggestion
  @Deactivate
  public synchronized void deactivate() {
    db.close();
  }

  @Reference(target = "(osgi.unit.name=org.opencast.annotation.impl.persistence)")
  public void setEntityManagerFactory(EntityManagerFactory entityManagerFactory) {
    this.entityManagerFactory = entityManagerFactory;
    resetDb();
  }

  @Reference
  public void setDBSessionFactory(DBSessionFactory dbSessionFactory) {
    this.dbSessionFactory = dbSessionFactory;
    resetDb();
  }
  
  private void resetDb() {
    db = dbSessionFactory.createSession(entityManagerFactory);
  }
```

feels like the more robust approach, but of course I don't know (much about) the internals of this new machinery, so maybe this is wrong for some reason? 🤔

I mean your version works, and I'm sure it also works in Opencast, probably because the two factories never change at runtime, but it's exactly assumptions like these that make hotloading some modules in Opencast break. 😔

Anyway, apparently neither of us knows better, so I won't press the issue. 🤷‍♀️ But if you want to investigate further, I will of course not stop you. 😉